routes/items.js

```diff
@@ -20,13 +20,45 @@ likely to add other models as routes are expanded upon
 
 const router = express.Router({ mergeParams: true });
 const jwtAuth = passport.authenticate('jwt', { session: false });
-
+const { moveItem, deleteItem } = require('./utils');
 router.use(express.json());
 router.use(jwtAuth);
 
 //Add an item to a list
 router
   .route('/')
+  .patch((req, res, next) => {
+    const { id: user } = req.user;
+    const { listId } = req.params;
+    const { direction, itemId } = req.body;
+    const requiredFields = ['direction', 'itemId'];
+    const missingField = requiredFields.find(field => !(field in req.body));
+    if (missingField) {
+      throw new ValidationError(missingField, 'Missing Field', 422);
+    }
+    if (!mongoose.Types.ObjectId.isValid(listId)) {
+      throw new HttpError(422, `${listId} is not a valid ObjectId`);
+    }
+
+    if (!mongoose.Types.ObjectId.isValid(itemId)) {
+      throw new HttpError(422, `${itemId} is not a valid ObjectId`);
+    }
+
+    ShoppingList.findOne({ user, _id: listId })
+      .then(list => {
+        let movedList;
+        if (direction === 'up') {
+          movedList = moveItem(list.items, itemId, list.head);
+        } else {
+          movedList = moveItem(list.items, itemId, list.head, true);
+        }
+        list.items = movedList.newItems;
+        list.head = movedList.head;
+        return list.save();
+      })
+      .then(newList => res.json(newList))
```

Comment:
I think for consistency, you should return an object like this:

```javascript
res.json({ list: newList })
```

Because for all other requests our responses our wrapped in an object with a key, right?